pkg/platform/kube/monitoring/test/function_test.go

```diff
@@ -205,14 +205,14 @@ func (suite *FunctionMonitoringTestSuite) TestNoRecoveryAfterDeployError() {
 			// wait for monitoring
 			time.Sleep(postDeploymentSleepInterval)
 
-			// ensure function is still in error state (due to deploy error of missing configmap)
-			suite.GetFunctionAndExpectState(getFunctionOptions, functionconfig.FunctionStateError)
+			// ensure function is still in error unhealthy (due to deploy error of wrong handler name)
```

Comment:
```suggestion
			// ensure function is still in unhealthy state (due to deploy error of wrong handler name)
```

Reply:
Why isn't this error though? What is the state of the pod in this case?

Reply:
@TomerShor it will be in a crash loop:

```
 NuclioFunction in unhealthy state:\n\nError - NuclioFunction pod (nuclio-function-no-recover-after-deploy-fail-6b94459d8f-fl9wv) is in a crash loop\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:2882\n\nCall stack:\nNuclioFunction pod (nuclio-function-no-recover-after-deploy-fail-6b94459d8f-fl9wv) is in a crash loop\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:2882\nNuclioFunction deployment failed\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:607\nFailed to wait for function deployment readiness\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:395\nFailed to wait for function resources to be available\n    .../platform/kube/controller/nucliofunction.go:263\n\n    .../nuclio/pkg/platform/kube/client/deployer.go:571\n\nCall stack:\nNuclioFunction in unhealthy state:\n\nError - NuclioFunction pod (nuclio-function-no-recover-after-deploy-fail-6b94459d8f-fl9wv) is in a crash loop\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:2882\n\nCall stack:\nNuclioFunction pod (nuclio-function-no-recover-after-deploy-fail-6b94459d8f-fl9wv) is in a crash loop\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:2882\nNuclioFunction deployment failed\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:607\nFailed to wait for function deployment readiness\n    .../nuclio/pkg/platform/kube/functionres/lazy.go:395\nFailed to wait for function resources to be available
 ```